### PR TITLE
[FLINK-4287] Ensure the yarn-session.sh classpath contains all Hadoop related paths

### PR DESCRIPTION
**Pull request by nielsbasjes**

### CODE REVIEW

File: flink-dist/src/main/flink-bin/yarn-bin/yarn-session.sh [hunk 1]
```diff
@@ -52,5 +52,5 @@ log_setting="-Dlog.file="$log" -Dlog4j.configuration=file:"$FLINK_CONF_DIR"/log4
 
 export FLINK_CONF_DIR
 
-$JAVA_RUN $JVM_ARGS -classpath $CC_CLASSPATH:$HADOOP_CLASSPATH:$HADOOP_CONF_DIR:$YARN_CONF_DIR  $log_setting org.apache.flink.yarn.cli.FlinkYarnSessionCli -j $FLINK_LIB_DIR/flink-dist*.jar "$@"
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/yarn-bin/yarn-session.sh]: Why did you change the qualified name of the `FlinkYarnSessionCli` ?

Comment (nielsbasjes) [inline flink-dist/src/main/flink-bin/yarn-bin/yarn-session.sh]: Thanks. Good catch. 
I changed the script in my installed 1.0.3 version and then committed that.
I missed the change in the classname.
As you can see I just fixed it.

**Comment by mxm** [comment]
Thanks @nielsbasjes. Will merge with #2320.

File: flink-dist/src/main/flink-bin/yarn-bin/yarn-session.sh (hunk 1 above)
Comment (mxm) [inline flink-dist/src/main/flink-bin/yarn-bin/yarn-session.sh]: The name was changed recently (some months ago). Niels was just on an old base.

**Comment by mxm** [comment]
Woops. Didn't end up merging #2320 so this got overlooked.

**Comment by mxm** [comment]
Rebased and ready to go. Merging.
